src/harness/reference_models/geo/utils.py

```diff
@@ -42,6 +42,48 @@ def GeoJsonToShapelyGeometry(geo):
   return geo
 
 
+def GridPolygon(poly, res_arcsec):
+  """Grids a polygon or multi-polygon.
+
+  This performs regular gridding of a polygon in PlateCarree (equirectangular)
+  projection (ie with fixed step in degrees in lat/lon space).
+  Points falling in the boundary of polygon will be included.
+
+  Args:
+    poly: A Polygon or MultiPolygon in WGS84 or NAD83, defined either
+      as a `shapely` or GeoJSON dict geometries.
+    res_arcsec: The resolution (in arcsec) used for regular gridding.
+
+  Returns:
+    A list of (lon, lat) defining the grid points.
+  """
+  if isinstance(poly, dict):
+    poly = GeoJsonToShapelyGeometry(poly)
+    return GridPolygon(poly, res_arcsec)
+
+  if isinstance(poly, sgeo.MultiPolygon):
+    # Optional block: for MultiPolygons, we process per polygon
+    # to avoid inefficiencies if polygons largely disjoint.
+    pts = ops.unary_union(
+        [sgeo.asMultiPoint(GridPolygon(p, res_arcsec))
+         for p in poly])
+    return [(p.x, p.y) for p in pts]
+
+  res = res_arcsec / 3600.
+  bounds = poly.bounds
+  lng_min = int(bounds[0] / res) * res
```

Comment:
I made the change, although it wont have any effect: 
because if the min of the division with the step (lat_min/res) is -100.5, it means you actually have to start putting a point at -100, and not at -101 which is outside of the polygon for sure.
But for the sake of extra safeguard, i did the change.

Reply:
It does have an effect. Say we have 360 arcsec resolution
if lng_min = -100.5, and lng_max = -100, int will be -100. thus skipping -100.5, .6, .7, .8, .9 
while floor, -101 -100 respectively. grid will include -100.5, .6, .7, .8, .9

Reply:
I think you got confused, by the meaning of the -100 in my example: it was not the latitude but the ratio of latitude and res: latitude/res.

Taking back your example of real_lng_min=-100.5 and real_lng_max=100, with res_arcsec = 360:

   res = 360/3600 = 1/10
Case int:
   lng_min = int(-100.5 * 10) * 1/10 = int(-1005) * 1/10 = -100.5
   => we start correctly with the -100.5 sample
Case floor:
  lng_min = floor(-100.5 * 10) * 1/10 = -100.5  
   => same result

Lets now take real_lng_min = -100.55. The samples shall still be -100.5, -100.6, ...
Case int:
  lng_min = int(-100.55*10) * 1/10 = int(-1005.5)  * 1/10 = -1005   *1/10. = -100.5
  => we start from the correct sample
Case floor
  lng_min = floor(100.55*10)*1/10 = floor(-1005.5) * 1/10 = -1006 * 1/10. = -100.6
 => we start with -100.6 which is obviously outside

Reply:
Thanks for the clarifications